### PR TITLE
Implement support for IAM Roles

### PR DESCRIPTION
**Pull request by zackproser**
## Description

These changes are based on #251, but also introduce concurrent deletion of roles for speed. They introduce support for inspecting and nuking IAM Roles. 

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

File: aws/iam_role.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	})
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return nil
+}
+
+// Nuke a single IAM Role
+func nukeRole(svc *iam.IAM, roleName *string) error {
```
Comment (yorinasub17) [inline aws/iam_role.go:121]: Looks like this is unused and replaced by `deleteIamRoleAsync`?

**Review by yorinasub17 — COMMENTED** [review]
Code mostly LGTM! Given the criticality of IAM Roles, I'd like to do a quick sanity check run with `go run` before approving, but I ran out of time for today. Will do that check tomorrow.

Also, looks like there are build failures for unit tests in CircleCI that should be fixed.

File: .circleci/nuke_config.yml [hunk 2]
```diff
@@ -106,3 +106,7 @@ IAMUsers:
     names_regex:
       # We want to make sure the main circle-ci-test user is never deleted
       - "^circle-ci-test$"
+IAMRoles:
+  include:
```
[the remark this answers is not in the document]
Comment (zackproser) [reply .circleci/nuke_config.yml:110]: Roger - sounds good!

File: aws/iam_role.go (hunk 1 above)
Comment (zackproser) [reply aws/iam_role.go]: Removed in f37f92a

**Comment by zackproser** [comment]
> Code mostly LGTM! Given the criticality of IAM Roles, I'd like to do a quick sanity check run with `go run` before approving, but I ran out of time for today. Will do that check tomorrow.
> 
> Also, looks like there are build failures for unit tests in CircleCI that should be fixed.

Tests are fixed. Looks like the two failing now are: 
- TestNukeTransitGateway 
- TestGetAllTransitGatewayVpcAttachment 

which are both marked as Flaky

**Review by yorinasub17 — APPROVED** [review]
LGTM! I also ran it locally and it worked as expected.

**Pull request merged**